Meadowlark-js/packages/meadowlark-core/src/model/DocumentIdentity.ts

```diff
@@ -82,6 +82,17 @@ export function meadowlarkIdForDocumentIdentity(
   return `${resourceInfoHashFrom(resourceInfo)}${documentIdentityHashFrom(documentIdentity)}` as MeadowlarkId;
 }
 
+/**
+ * Generate a UUID v4
+ */
 export function generateDocumentUuid(): DocumentUuid {
   return crypto.randomUUID() as DocumentUuid;
 }
+
+/**
+ * Check that this is a well formed UUID v4
+ */
+export function isDocumentUuidWellFormed(documentUuid: DocumentUuid): boolean {
+  const uuid4Regex = /^[0-9a-f]{8}-[0-9a-f]{4}-[4][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/;
```

Comment:
Would it make sense to move this constant into the module scope so that the reg ex object does not need to be recreated with each access?

Reply:
Ouch, yes